src/controller/CHIPDeviceController.cpp

```
@@ -894,8 +894,19 @@ CHIP_ERROR DeviceCommissioner::StopPairing(NodeId remoteDeviceId)
 
 CHIP_ERROR DeviceCommissioner::UnpairDevice(NodeId remoteDeviceId)
 {
-    // TODO: Send unpairing message to the remote device.
-    return CHIP_NO_ERROR;
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    MATTER_TRACE_EVENT_SCOPE("UnpairDevice", "DeviceCommissioner");
+
+    if (mState == State::Initialized)
+    {
+        err = AutoCurrentFabricRemover::RemoveCurrentFabric(this, remoteDeviceId);
+    }
+    else // the device is currently being paired
+    {
+        err = StopPairing(remoteDeviceId);
```

Why not just return an error?

---

I will change to return error directly.